Lib/test/test_os.py

```diff
@@ -3102,6 +3102,18 @@ def test_dup2(self):
         self.assertEqual(os.dup2(fd, fd3, inheritable=False), fd3)
         self.assertFalse(os.get_inheritable(fd3))
 
+        # dup2(fd, fd) must have no effect for a valid fd
```

Comment:
Please move this test into a separated method and emit somehow a SkipTest exception to record that the test is skipped on old Android versions.